ext/hash/hash.c

```diff
@@ -1529,8 +1555,15 @@ PHP_METHOD(HashContext, __unserialize)
 
 	hash->ops = ops;
 	hash->context = php_hash_alloc_context(ops);
-	ops->hash_init(hash->context);
 	hash->options = options;
+	if (args_zv && IS_ARRAY == Z_TYPE_P(args_zv) && zend_hash_num_elements(Z_ARRVAL_P(args_zv)) > 0) {
+		ops->hash_init(hash->context, Z_ARRVAL_P(args_zv));
+		hash->args = Z_ARRVAL_P(args_zv);
+		GC_ADDREF(hash->args);
```

Comment:
Probably interned arrays cannot occur here, but I'd still use GC_TRY_ADDREF to be safe.

Reply:
Fixed, thanks.